src/AspNetCoreModuleV2/AspNetCore/src/applicationinfo.cpp

```diff
@@ -54,42 +38,103 @@ APPLICATION_INFO::~APPLICATION_INFO()
 HRESULT
 APPLICATION_INFO::Initialize(
     _In_ IHttpServer              *pServer,
-    _In_ IHttpApplication         *pApplication,
-    _In_ FILE_WATCHER             *pFileWatcher
+    _In_ IHttpApplication         *pApplication
 )
 {
     HRESULT hr = S_OK;
 
     DBG_ASSERT(pServer);
     DBG_ASSERT(pApplication);
-    DBG_ASSERT(pFileWatcher);
 
     // todo: make sure Initialize should be called only once
     m_pServer = pServer;
     FINISHED_IF_NULL_ALLOC(m_pConfiguration = new ASPNETCORE_SHIM_CONFIG());
     FINISHED_IF_FAILED(m_pConfiguration->Populate(m_pServer, pApplication));
     FINISHED_IF_FAILED(m_struInfoKey.Copy(pApplication->GetApplicationId()));
-    FINISHED_IF_NULL_ALLOC(m_pFileWatcherEntry = new FILE_WATCHER_ENTRY(pFileWatcher));
-
-    UpdateAppOfflineFileHandle();
 
 Finished:
     return hr;
 }
 
-HRESULT
-APPLICATION_INFO::StartMonitoringAppOffline()
+BOOL
+APPLICATION_INFO::AppOfflineFound()
 {
-    if (m_pFileWatcherEntry != NULL)
+    ULONGLONG  ulCurrentTime = GetTickCount64();
+    //
+    // we only care about app offline presented. If not, it means the application has started
+    // and is monitoring  the app offline file
+    // we cache the file exist check for 1 minute
+    //
+    if (m_fAppOfflineFound && ulCurrentTime - m_ulLastCheckTime > 1000)
```

Comment:
This would not check app_offline on the first request if ulCurrentTime - m_ulLastCheckTime < 1000. For example, if have app_offline already in the folder and then send a request, it will not check for app_offline even though it should.

Reply:
m_ulLastCheckTime is 0, ulCurrentTime is current tick number since machine reboot which cannot be less than 1000 (i.e., 1s)

Reply:
> we cache the file exist check for 1 minute

One second?

Reply:
yes. one second. a typo